src/Test.UnitTests.Sarif/GitHelperTests.cs

```diff
@@ -118,6 +118,49 @@ public void GetRepositoryRoot_WhenCalledOnTheDefaultInstanceWithCachingDisabled_
             action.Should().NotThrow();
         }
 
+        [Fact]
+        public void GetGitExePath_WhenPathExistsInProgramFiles()
+        {
+            var mockFileSystem = new Mock<IFileSystem>();
+
+            mockFileSystem.Setup(x => x.FileExists(It.IsAny<string>())).Returns(true);
+
+            var gitHelper = new GitHelper(mockFileSystem.Object);
+
+            gitHelper.GetGitExePath().Should().NotBeNullOrEmpty();
+        }
+
+        [Fact]
+        public void GetGitExePath_WhenPathDoesNotExistInProgramFiles()
+        {
+            var mockFileSystem = new Mock<IFileSystem>();
+
+            mockFileSystem.Setup(x => x.FileExists(It.IsAny<string>())).Returns(false);
+
+            var gitHelper = new GitHelper(mockFileSystem.Object);
+
+            gitHelper.GetGitExePath().Should().NotBeNull();
+        }
+
+        [Fact]
+        public void SearchForFileInEnvironmentVariable_WhenVariableDoesNotExist()
+        {
+            FileSearcherHelper.SearchForFileInEnvironmentVariable("PATH_THAT_DOES_NOT_EXIST", "filename.exe").Should().BeNull();
+        }
+
+        [Fact]
+        public void SearchForFileInEnvironmentVariable_WhenVariableExistsButFileDoesnt()
+        {
+            // The error in the ntdll name here is intentional.
+            FileSearcherHelper.SearchForFileInEnvironmentVariable("PATH", "ntdll.dlll").Should().BeNull();
+        }
+
+        [Fact]
+        public void SearchForFileInEnvironmentVariable_WhenVariableAndFileExists()
+        {
+            FileSearcherHelper.SearchForFileInEnvironmentVariable("PATH", "ntdll.dll").Should().NotBeNull();
```

Comment:
>ntdll.dll [](start = 75, length = 9)

This test will fail on non-Windows platforms. @michaelcfaning @eddynaka

Reply:
I filed [https://github.com/microsoft/sarif-sdk/issues/2113](https://github.com/microsoft/sarif-sdk/issues/2113): Unit test fails on non-Windows platforms.

---
In reply to: [505655086](https://github.com/microsoft/sarif-sdk/pull/2107#discussion_r505655086) [](ancestors = 505655086)